libres/lib/analysis/update.cpp

```diff
@@ -445,14 +446,33 @@ void assert_size_equal(int ens_size, const bool_vector_type *ens_mask) {
             ", mask_size:" + std::to_string(bool_vector_size(ens_mask)));
 }
 
+// Opens and returns a log file.
+static FILE *log_step_list(const char *log_path,
+                           const std::vector<int> &step_list) {
+    char *log_file;
+    if (step_list.size() == 1)
+        log_file = util_alloc_sprintf("%s%c%04d", log_path, UTIL_PATH_SEP_CHAR,
+                                      step_list.front());
+    else
+        log_file =
+            util_alloc_sprintf("%s%c%04d-%04d", log_path, UTIL_PATH_SEP_CHAR,
+                               step_list.front(), step_list.back());
+    FILE *log_stream = util_fopen(log_file, "w");
+
+    free(log_file);
+    return log_stream;
```

Comment:
Since we're touching this already...

```suggestion
    std::string log_file;
    if (step_list.size() == 1)
        log_file = fmt::format("{}{}{:04d}", log_path, UTIL_PATH_SEP_CHAR,
                                      step_list.front());
    else
        log_file =
            util_alloc_sprintf("{}{}{:04d}-{:04d}", log_path, UTIL_PATH_SEP_CHAR,
                               step_list.front(), step_list.back());
    FILE *log_stream = fopen(log_file.data(), "w");
    if (log_stream == nullptr)
        throw std::runtime_error(fmt::format("Error opening '{}' for writing: {}", log_file, strerror(errno)));
    return log_stream;
```

(Might have to include `cerrno` and `fmt/format.h`)